File: Views/Components/WidgetPreviewSection.swift
import SwiftUI

struct WidgetPreviewSection: View {
    let onWidgetTap: () -> Void
    
    var body: some View {
        Button(action: onWidgetTap) {
            HStack(spacing: 16) {
                // Contenu principal
                VStack(alignment: .leading, spacing: 6) {
                    // Titre principal
                    Text(ui: "add_widgets", comment: "Add widgets title")
                        .font(.system(size: 22, weight: .semibold))
                        .foregroundColor(.black)
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 20)
                    
                    // Sous-titre
                    Text(ui: "feel_closer_partner", comment: "Feel closer partner subtitle")
                        .font(.system(size: 16))
                        .foregroundColor(.black.opacity(0.7))
                        .multilineTextAlignment(.center)
                        .padding(.horizontal, 20)
                }
                
                Spacer()
                
                // Icône à droite
                Image(systemName: "chevron.right")
                    .font(.system(size: 14))
                    .foregroundColor(.black.opacity(0.5))
            }
            .padding(.horizontal, 24)
            .padding(.vertical, 20)
            .background(
                RoundedRectangle(cornerRadius: 16)
                    .fill(Color.white.opacity(0.95))
                    .shadow(color: Color.black.opacity(0.1), radius: 8, x: 0, y: 2)
            )
        }
        .buttonStyle(PlainButtonStyle())
        .padding(.horizontal, 20)
    }
}

struct WidgetPreviewSection_Previews: PreviewProvider {
    static var previews: some View {
        WidgetPreviewSection(onWidgetTap: {
            print("Widget card tapped!")
        })
        .padding()
        .background(Color.purple)
    }
} 

File: Views/Components/WidgetPreviewSection.kt
// WidgetPreviewSection.kt
package your.package.name

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ChevronRight
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun WidgetPreviewSection(
    onWidgetTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    Surface(
        onClick = onWidgetTap,
        shape = androidx.compose.foundation.shape.RoundedCornerShape(16.dp),
        color = Color.White.copy(alpha = 0.95f),
        shadowElevation = 8.dp,
        modifier = modifier
            .padding(horizontal = 20.dp)
            .fillMaxWidth()
    ) {
        Row(
            modifier = Modifier
                .padding(horizontal = 24.dp, vertical = 20.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                // Titre principal
                Text(
                    text = stringResource(R.string.add_widgets),
                    fontSize = 18.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color.Black,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis
                )
                // Sous-titre
                Text(
                    text = stringResource(R.string.feel_closer_partner),
                    fontSize = 14.sp,
                    color = Color.Black.copy(alpha = 0.7f),
                    maxLines = 3,
                    overflow = TextOverflow.Ellipsis
                )
            }

            Spacer(modifier = Modifier.width(16.dp))

            // Icône chevron (auto-miroir pour RTL)
            Icon(
                imageVector = Icons.AutoMirrored.Filled.ChevronRight,
                contentDescription = null,
                tint = Color.Black.copy(alpha = 0.5f),
                modifier = Modifier.size(14.dp)
            )
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFF800080)
@Composable
private fun WidgetPreviewSectionPreview() {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF800080))
            .padding(16.dp)
    ) {
        WidgetPreviewSection(onWidgetTap = { /* println("Widget card tapped!") */ })
    }
}
